server/services/timelineService.ts

```diff
@@ -21,6 +21,29 @@ export default class TimelineService {
       const timelineResponse = await this.educationAndWorkPlanClient.getTimeline(prisonNumber, token)
       timelineResponse.events = timelineResponse.events.filter(this.filterTimelineEvents)
 
+      // Check for GOAL_CREATED events with the same correlationId, then change the event type to MULTIPLE_GOALS_CREATED
```

Comment:
It's cool that it's working 👍  Nice work 👍 

though I am struggling to read and understand this new block of code that does the magic 😁  I also notice this new code is not covered by any new tests

I wonder if we have some options here .... maybe rework this code so it's easier to follow? (and include tests for it in the main timelineService.test.ts); or factor it out into some external function that can be unit tested independently (and mocked in timelineService.test.ts)
Or even consider whether this is even the right place to do this ... 🤔  .... I'm wondering if this grouping is more of a UI concern than this main service and mapping, and perhaps it would be better placed somewhere closer to the view? Perhaps in the controller method, or maybe in the View object?
I don't know the answer to the above, but I'm not sure what we have here is quite right yet ... 🤔 

What are you thoughts about it?